src/mds/SnapRealm.cc

```diff
@@ -435,7 +436,8 @@ void SnapRealm::build_snap_trace() const
 
     dout(10) << "build_snap_trace my_snaps " << info.my_snaps << dendl;
 
-    SnapRealmInfoNew ninfo(info, srnode.last_modified, srnode.change_attr);
+    SnapRealmInfoNew ninfo(info, srnode.last_modified,
+                           srnode.change_attr, srnode.is_snapdir_visible());
```

Comment:
Perhaps pass `srnode` and let `SnapRealmInfoNew` grab what it wants.

Reply:
we only need the new snaprealm metadata here, passing entire sr_t would be an overkill.